### PR TITLE
feat: allow customizable database engine

### PR DESCRIPTION
**Pull request by fmartingr**
#### Summary

- Renames `db_engine_version` to `db_default_engine_version`
- Adds new `db_engine_version` (customizable, string)
- Adds new parameter to `DeployerConfig.TerraformDBSettings.EngineVersion`
- Logic now uses the above variable is set, defaults otherwise

### CODE REVIEW

File: config/deployer.sample.json [hunk 1]
```diff
@@ -47,7 +47,8 @@
     "Password": "mostest80098bigpass_",
     "EnablePerformanceInsights": true,
     "DBParameters": [],
-    "ClusterIdentifier": ""
+    "ClusterIdentifier": "",
+    "EngineVersion": ""
```
Comment (agnivade) [inline config/deployer.sample.json:51]: Let's have a default value here.

File: deployment/config.go [hunk 2]
```diff
@@ -170,6 +170,8 @@ type TerraformDBSettings struct {
 	// If ClusterIdentifier is not empty, DBName should be set to the name of the database in such cluster.
 	// If ClusterIdentifier is empty, the database created will use DBName as its name.
 	DBName string `default:""`
+	// EngineVersion specifies the version of the database engine.
+	EngineVersion string `default:""`
```
Comment (agnivade) [inline deployment/config.go:174]: Have a default value here.

And also, I'd like to have some validation where the engine version is matched to a set of valid version strings depending on the engine type. AWS will automatically throw an error if there's a mistake, so it's better to do some checking from our side.

Comment (fmartingr) [reply deployment/config.go:174]: Should I go ahead and default targeting postgres only?

**Comment by wiggin77** [comment]
@agnivade @streamer45 can we put a priority on reviewing this and https://github.com/mattermost/mattermost-load-test-ng/pull/810?  There is customer sensitivity around timing.

File: deployment/config.go (hunk 2 above)
Comment (agnivade) [reply deployment/config.go:189]: I think that should be fine.

**Comment by streamer45** [comment]
> @agnivade @streamer45 can we put a priority on reviewing this and #810? There is customer sensitivity around timing.

@wiggin77 @fmartingr Happy to review, but for anything dead urgent, I'd think a custom branch would work best. Or is there a need to merge to master in such a hurry?

**Comment by wiggin77** [comment]
> > @agnivade @streamer45 can we put a priority on reviewing this and #810? There is customer sensitivity around timing.
> 
> @wiggin77 @fmartingr Happy to review, but for anything dead urgent, I'd think a custom branch would work best. Or is there a need to merge to master in such a hurry?

Great point @streamer45.  @fmartingr can you create a feature branch for this customer effort that will get merged to master later?  Both these PRs and any other related PRs will be merged to the feature branch for now.

**Comment by fmartingr** [comment]
> > > @agnivade @streamer45 can we put a priority on reviewing this and #810? There is customer sensitivity around timing.
> > 
> > 
> > @wiggin77 @fmartingr Happy to review, but for anything dead urgent, I'd think a custom branch would work best. Or is there a need to merge to master in such a hurry?
> 
> Great point @streamer45. @fmartingr can you create a feature branch for this customer effort that will get merged to master later? Both these PRs and any other related PRs will be merged to the feature branch for now.

We already have one. We wanted to be more clean this time (merging only commits from upstream) but I'm happy to go that route.

File: deployment/config.go (hunk 2 above)
Comment (fmartingr) [reply deployment/config.go:189]: So, thinking about this a bit more. If I set up a default value here the actual defaults are not going to be used:

```tf
  engine_version      = var.db_engine_version != "" ? var.db_engine_version : var.db_default_engine_version[var.db_instance_engine]
```

File: config/deployer.sample.json (hunk 1 above)
Comment (fmartingr) [reply config/deployer.sample.json:51]: Leaving it empy would pick the default value from the terraform variable:

```tf
variable "db_default_engine_version" {
  type = map(any)
  default = {
    "aurora-mysql"      = "8.0.mysql_aurora.3.05.2"
    "aurora-postgresql" = "14.7"
  }
}
```

Comment (sadohert) [reply config/deployer.sample.json:51]: Low urgency here, but I think putting the default into the sample is helpful for the operator.

**Comment by fmartingr** [comment]
Replaced by #814

**Pull request closed without merging**